site/content@beta/backend-apis/core-info.md

```diff
@@ -0,0 +1,155 @@
+---
+weight: 30
+---
+
+# Key concepts
+
+## Chats and threads
+
+By looking at the chat structure, you notice that each **chat** is is divided into **threads**. Every thread contains **events**, for example sent messages. You can think of a **chat** as **a whole conversation**, while **threads** are separate **conversation topics**.
+
+![Chats and Threads](chat_threads.png "chats and threads")
+
+Consider the example of an online store. A customer starts a chat to ask about the shoes he bought. If he hasn't got any previous chat history, **a new chat is started**. Within this chat, **a new thread is created** (conversation topic). The messages he exchanges with the customer service assistant come to the thread as **events**. After solving the problem, the custmer says goodbye and ends the chat, which automatically **closes the thread**. Let's say the customer is rude and leaves without saying goodbye. In this case, the thread closes after 30 minutes of inactivity (time periods are configurable). The chat **doesn't end**, though.
+
+The next day, customer returns to ask about his other purchase. A new thread starts in the context of the chat, which has been **continuously open**. Each time customer returns and starts a new conversation topic, he needs to click **_start a chat_**. If we stuck to our naming convention, that would have to be **_start a thread_**, because this is what he actually does behind the scenes. 
+
+A **message** is not the only event type, though. To check out other event types, see the [Events](#events) section.
+
+
+### Rules and conditions
+
+Here are some general rules, which summarize the previous section and add new info.
+
+1. When a new chat is started, **a new thread** is created within this chat. New threads within a single chat are created on the server side. 
+
+2. There's always **only one active thread**. The **last thread** is always the **active** one. Events are always added to the active thread. 
+
+3.  When there's no active thread in the chat, sending an event starts **a new thread**. The [annotation event](#annotation) is an exception; it is added at the end of the **last thread**, instead of starting a new thread.
+
+4. There can be time gaps between threads in a chat, but once a chat is started, it's **continuously open**.
+
+5. Customers can send messages even when agents are **offline**.
+
+6. **Multiple users** (Agents or Customers) can participate in a single chat. 
```

Comment:
Obecnie może być tylko 1 customer w 1 czacie i wielu agentów (w przyszłości zdejmiemy to ograniczenie)